### ISSUE TITLE
A bunch of people are seeing content length issues when updating with composer

### ISSUE THREAD

**Issue by marcgetter** [hiqdev/asset-packagist#105]
### Please provide your `composer.json` if appropriate.

When running 

composer create-project --no-interaction acquia/blt-project project-name,

gettiing this error.

Content-Length mismatch, received 130816 bytes out of the expected 3999824

It looks like this file is getting cut off:

https://asset-packagist.org/p/provider-latest/32025501a5070bb135cc140c5bdfa3e3375bff8b839a7f6d5e3f52fb68bb32b1.json

**Comment by bkosborne** [comment]
Same. Came here to report this. When I open the file in FireFox I get this error: `SyntaxError: JSON.parse: unterminated string at line 1 column 392660 of the JSON data`

**Comment by timcoxopc** [comment]
I am getting a similar issue with: 

https://asset-packagist.org/p/provider-latest/46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json

Content-Length mismatch, received 65280 out of 3999824 bytes

The file is also cut off when viewing the .json file.

**Comment by sanovskiy** [comment]
> I am getting a similar issue with:
> 
> https://asset-packagist.org/p/provider-latest/46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json
> 
> Content-Length mismatch, received 65280 out of 3999824 bytes
> 
> The file is also cut off when viewing the .json file.

Wget tries to download partial
```$ wget https://asset-packagist.org/p/provider-latest/46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json
--2019-10-28 11:09:13--  https://asset-packagist.org/p/provider-latest/46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json
Resolving asset-packagist.org (asset-packagist.org)... 185.56.232.198
Connecting to asset-packagist.org (asset-packagist.org)|185.56.232.198|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 3999824 (3.8M) [application/json]
Saving to: '46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json.1'

46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f   2%[=>                                                                                                                                       ]  79.75K  --.-KB/s    in 0.1s    

2019-10-28 11:09:14 (735 KB/s) - Connection closed at byte 81659. Retrying.

--2019-10-28 11:09:15--  (try: 2)  https://asset-packagist.org/p/provider-latest/46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json
Connecting to asset-packagist.org (asset-packagist.org)|185.56.232.198|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 3999824 (3.8M), 3918165 (3.7M) remaining [application/json]
Saving to: '46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json.1'

46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f   4%[++==>                                                                                                                                    ] 159.46K  --.-KB/s    in 0.1s    

2019-10-28 11:09:15 (735 KB/s) - Connection closed at byte 163283. Retrying.

--2019-10-28 11:09:17--  (try: 3)  https://asset-packagist.org/p/provider-latest/46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json
Connecting to asset-packagist.org (asset-packagist.org)|185.56.232.198|:443... connected.
HTTP request sent, awaiting response... 206 Partial Content
Length: 3999824 (3.8M), 3836541 (3.7M) remaining [application/json]
Saving to: '46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f7fbfc.json.1'

46dd90034cc9b9f8835f29c76309a317e62c99f35bbfdb43c128faab43f   6%[+++++==>                                                                                                                                 ] 239.17K  --.-KB/s    in 0.1s    

2019-10-28 11:09:18 (798 KB/s) - Connection closed at byte 244906. Retrying.
```

**Comment by DmitryBay** [comment]
Have same problem also.

**Comment by HenryVolkmer** [comment]
confirmed, json invalid. 
`As a workaround, if you are not actually updating the packages you are requiring from asset-packagist, you can just remove the repo from your composer.json while you are updating.`

https://www.bountysource.com/issues/82594637-content-length-mismatch

**Comment by podarok** [comment]
Same for 
https://asset-packagist.org/p/provider-latest/b5908562564adbe06c4e702fd75e7faa8f6d9e7a08a2f99b551bfdcb1300f79d.json

`Content-Length mismatch, received 130816 out of 4000042 bytes:`

**Comment by demonking** [comment]
#104

**Comment by SilverFire** [comment]
Working on the problem, thanks for the update

**Comment by SilverFire** [comment]
Fixed. Sorry for the inconvenience.

**Comment by larryli** [comment]
composer update failed for download `https://asset-packagist.org/p/provider-latest/e466e847c6e58fc7659581a46582a2fc80a17fc843cb977a682e1de70e9b1eb3.json`

**Comment by mv4** [comment]
I am experiencing the same problem today.

**Comment by yusupovb** [comment]
Same problem

[Composer\Downloader\TransportException]                                                                                                      
Content-Length mismatch, received 81664 bytes out of the expected 4000651

**Comment by demonking** [comment]
Are you sure this problem still happens?
I have tried it, and it works fine

**Comment by yusupovb** [comment]
It happens time to time.

**Comment by alexandrdnlv** [comment]
One more example:
```
Downloading https://asset-packagist.org/p/provider-latest/34547facb9e3710779c9d8107adfa7313d282862cc1c58f0d2a640c5a132eeb2.json
Content-Length mismatch, received 130816 out of 4000762 bytes:
```

**Comment by Tawreh** [comment]
Seeing this error every time I try to install or update Composer packages currently. I've cleared composer caches. Is there some kind of global TTL that we need to wait for? (I'm UK-based, if it makes a difference.)

**Comment by demonking** [comment]
@SilverFire what was the problem the last times?

**Comment by Xarno** [comment]
Happens now again.
https://asset-packagist.org/p/provider-latest/34547facb9e3710779c9d8107adfa7313d282862cc1c58f0d2a640c5a132eeb2.json is incomplete.

**Comment by SilverFire** [comment]
The additional storage was purchased but has not been mounted yet.
The reboot is scheduled on this evening

**Comment by SilverFire** [comment]
Meanwhile, the problem is fixed manually again. Sorry, guys, I hope this evening reboot will be the last outage.

**Comment by Tawreh** [comment]
Thank you for the update :) I'll stay away from Composer today and update things tomorrow then!

**Comment by SilverFire** [comment]
![image](https://user-images.githubusercontent.com/4499203/67760888-d20b7180-fa4a-11e9-958c-d1a13c101b2c.png)

Now we have a lot of free space.

**Comment by kdo70** [comment]
I am experiencing the same problem today.

```
  [Composer\Downloader\TransportException]                                   
  Content-Length mismatch, received 81664 bytes out of the expected 4075770  
```

**Comment by authintmedia** [comment]
Same here, this issue has been hitting hard lately. For past week now.

`[Composer\Downloader\TransportException]                                   
  Content-Length mismatch, received 98048 bytes out of the expected 4075770 `

**Comment by SilverFire** [comment]
No way... 

![image](https://user-images.githubusercontent.com/4499203/67920675-c4223180-fbad-11e9-892b-5b91c37360cf.png)

```
139G    ./git-github.com-DefinitelyTyped-DefinitelyTyped.git
```

I could not imagine such a huge repo could be cloned

**Comment by SilverFire** [comment]
Btw, here is a list of top repositories by size.

```
139G    ./git-github.com-DefinitelyTyped-DefinitelyTyped.git
7.0G    ./git-github.com-expo-expo.git
3.1G    ./git-github.com-gatsbyjs-gatsby.git
2.9G    ./git-github.com-Microsoft-TypeScript.git
1.1G    ./git-github.com-mrdoob-three.js.git
925M    ./git-github.com-twitter-bootstrap.git
818M    ./git-github.com-oliver-moran-jimp.git
799M    ./git-github.com-twitter-twemoji.git
702M    ./git-github.com-storybookjs-storybook.git
617M    ./git-github.com-telusdigital-tds.git
555M    ./git-github.com-syncfusion-ej2-javascript-ui-controls.git
542M    ./git-github.com-ianstormtaylor-slate.git
455M    ./git-github.com-rse-typopro-web.git
448M    ./git-github.com-ICIJ-node-tika.git
380M    ./git-ssh---git-github.com-swagger-api-swagger-ui.git
374M    ./git-github.com-tensorflow-tfjs-core.git
352M    ./git-github.com-OfficeDev-office-ui-fabric-react.git
351M    ./git-github.com-telus-tds-core.git
347M    ./git-github.com-telusdigital-tds-core.git
297M    ./git-github.com-cerner-terra-core.git
279M    ./git-github.com-jupyterlab-jupyterlab.git
262M    ./git-ssh---git-github.com-swagger-api-swagger-editor.git
260M    ./git-github.com-transloadit-uppy.git
252M    ./git-github.com-angular-angular.git
250M    ./git-github.com-fex-team-webuploader.git
```

I have dropped the giant, so we have again a lot of free space. I also will consider purchasing additional 200GB to handle this mega-hyper-huge typescript repo.

![image](https://user-images.githubusercontent.com/4499203/67921445-63e0bf00-fbb0-11e9-9447-024160358b09.png)

I am very sorry this problem bothered you, guys :(

**Comment by authintmedia** [comment]
@SilverFire Thank you for such service, the system is back. Really appreciate all your hard work.

**Comment by demonking** [comment]
@SilverFire But don't unterstand, i though this is a composer+bower solution.

The Package https://github.com/fex-team/webuploader is an ActionScript(Flash)

**Comment by Dominus77** [comment]
This problem appeared again today
```
[Composer\Downloader\TransportException]
Content-Length mismatch, received 32512 bytes out of the expected 5542180
```
`Composer version 1.10.13 2020-09-09 11:46:34`

**Comment by Dominus77** [comment]
> This problem appeared again today
> 
> ```
> [Composer\Downloader\TransportException]
> Content-Length mismatch, received 32512 bytes out of the expected 5542180
> ```
> 
> `Composer version 1.10.13 2020-09-09 11:46:34`

The problem went away, worked without errors

**Comment by amjad1233** [comment]
I am experiencing the problem today :
` 18: transfer closed with 5563326 bytes remaining to read:	https://asset-packagist.org/p/provider-latest/113b5794ba27dd6964794393fee2e43aea85f1004344aea715f35fa6708ce5f1.json
    Finished: success: 0, skipped: 0, failure: 1, total: 1

In RemoteFilesystem.php line 345:

  Content-Length mismatch, received 130816 bytes out of the expected 5694137
`

**Comment by ashrafakl** [comment]
> I am experiencing the problem today :
> 
> ```
> 18: transfer closed with 5563326 bytes remaining to read:	https://asset-packagist.org/p/provider-latest/113b5794ba27dd6964794393fee2e43aea85f1004344aea715f35fa6708ce5f1.json
>     Finished: success: 0, skipped: 0, failure: 1, total: 1
> 
> In RemoteFilesystem.php line 345:
> 
>   Content-Length mismatch, received 130816 bytes out of the expected 5694137
> ```

Same error today
